File: 0300_Longest Increasing Subsequence/LongestIncreasingSubsequence_DP.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        if(nums == null || nums.length == 0) return 0;
        if(nums.length == 1) return 1;
        int ans = 1;
        int[] dp = new int[nums.length];
        
        for(int i=0;i<nums.length;i++){
            dp[i] = 1;
            for(int j=i-1;j>=0;j--){
                if(nums[i] > nums[j]){
                    dp[i] = Math.max(dp[i], dp[j]+ 1);
                    ans = Math.max(ans, dp[i]);
                }
            }
        }
        
        return ans;
    }
}

File: 0300_Longest Increasing Subsequence/LongestIncreasingSubsequence_DP.cs
public class Solution {
    public int LengthOfLIS(int[] nums) {
        var dp = Enumerable.Repeat(1, nums.Length).ToArray();
        var ans = 0;
        
        for(int i=0;i<nums.Length;i++) 
        {
            for(int j=i-1;j>=0;j--)
            {
                if(nums[i] > nums[j])
                    dp[i] = Math.Max(dp[i], dp[j] + 1);
            }
            
            ans = Math.Max(ans, dp[i]);
        }
        
        return ans;
    }
}